### PR TITLE
Add authentication docs

### PR DESCRIPTION
**Pull request by odlp**
Includes examples for Clearance, Devise and HTTP basic auth.

Also tidies up references to the old docs app which redirects to the prototype.

### CODE REVIEW

**Comment by BenMorganIO** [comment]
This PR is awesome! Thank you @odlp

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ Restart your server, and visit http://localhost:3000/admin
 to see your new dashboard in action.
 
 To customize the appearance, behavior, and contents of the dashboard,
-see the guides at http://administrate-docs.herokuapp.com.
+see the guides at http://administrate-prototype.herokuapp.com.
```
Comment (nickcharlton) [inline README.md]: Could we set the prototype links to all be `https`?

**Comment by nickcharlton** [comment]
One comment, but otherwise LGTM!

File: README.md (hunk 1 above)
Comment (odlp) [inline README.md]: Good spot! Will do

**Pull request merged**